Provider/src/FirebirdSql.Data.FirebirdClient/Common/BlobBase.cs

```diff
@@ -130,7 +127,7 @@ public void Write(byte[] buffer, int index, int count)
 
 		protected abstract void Create();
 		protected abstract void Open();
-		protected abstract byte[] GetSegment();
+		protected abstract void GetSegment(MemoryStream ms);
```

Comment:
Maybe just use plain `Stream`, because nothing requires the parameter to be `MemoryStream` and I'd call the parameter `stream`.

Reply:
Yes your right, its better that way, refactored in following commit